### PR TITLE
ssh-askpass-mac: update to 3.0

### PR DESCRIPTION
**Pull request by huyz**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.4 23E214 arm64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by huyz** [comment]
Anyone know how to obtain the `/opt/local/var/macports/logs/_Users_runner_work_macports-ports_macports-ports_ports_aqua_ssh-askpass-mac/ssh-askpass-mac/main.log` from the GitHub action run?
I don't have an Intel MacBook running macOS 13 to repro this build error.

Btw, I created an issue asking the developer for some insight as to what looks like a build error that seems to afflict only macOS 13 Intel: https://github.com/lukas-zronek/ssh-askpass-mac/issues/5

**Comment by ryandesign** [comment]
> Anyone know how to obtain the `/opt/local/var/macports/logs/_Users_runner_work_macports-ports_macports-ports_ports_aqua_ssh-askpass-mac/ssh-askpass-mac/main.log` from the GitHub action run?

Click the Details link next to the failed macos-13 build. Click the gear icon at the upper right. Click Download log archive.

It looks like the error is:

```
error: failed to deserialize Info.plist task context: Deserialization failed: SDK lookup failed for canonical name: macosx13.3 (in target 'ssh-askpass' from project 'ssh-askpass')
```

which is something I've seen with other ports (e.g. https://trac.macports.org/ticket/68417) and have not made an attempt to understand.

**Comment by huyz** [comment]
Thanks for the reply and review!

> Click the Details link next to the failed macos-13 build. Click the gear icon at the upper right. Click Download log archive.

That seems to give me the same logs I can see in the GitHub Action run UI on the web. But those logs refer to another log file: `/opt/local/var/macports/logs/_Users_runner_work_macports-ports_macports-ports_ports_aqua_ssh-askpass-mac/ssh-askpass-mac/main.log` which I was hoping would give even more details.  That's the file I'm still looking for. But perhaps that log file contains the same thing that's already caught via stdout/stderr and displayed in the GitHub Action run log?

> something I've seen with other ports (e.g. https://trac.macports.org/ticket/68417) 

Interesting. Great of you to notice the commonality.

File: aqua/ssh-askpass-mac/Portfile [hunk 1]
```diff
@@ -17,10 +17,11 @@ long_description        ssh-askpass-mac is a graphical front-end for ssh-add on
 use_xcode               yes
 
 github.tarball_from     archive
+distname                v${version}
```
[the remark this answers is not in the document]
Comment (huyz) [reply aqua/ssh-askpass-mac/Portfile]: Fixed in https://github.com/macports/macports-ports/pull/23181/commits/0223603c328d455915c3ef99bd5db480a26e380f

File: aqua/ssh-askpass-mac/Portfile [hunk 2]
```diff
@@ -38,14 +39,19 @@ post-destroot {
 notes "
 Launch ${name} with
     launchctl load -w \\
-        /Library/LaunchAgents/org.machports.ssh-askpass-mac.plist
+        /Library/LaunchAgents/org.macports.ssh-askpass-mac.plist
 
-On macOS < 12 (Monterey) or OpenSSH < 8.3 add the following line to the
-startup file of your shell:
+On macOS < 12 (Monterey) or OpenSSH < 8.4 add the following line to the startup
+file of your shell (~/.bash_profile (bash) or ~/.zprofile (zsh)):
```
[the remark this answers is not in the document]
Comment (huyz) [reply aqua/ssh-askpass-mac/Portfile]: Fixed in https://github.com/macports/macports-ports/pull/23181/commits/0223603c328d455915c3ef99bd5db480a26e380f

File: aqua/ssh-askpass-mac/Portfile [hunk 3]
```diff
@@ -38,14 +39,19 @@ post-destroot {
...
-    ssh-add $@ < /dev/null
+    ssh-add()
+    {
+        command ssh-add $@ </dev/null
+    }
 
-You can also disable the keychain with the following command:
+The setting 'Remember passphrase in my keychain' is enabled by default. A
+change of the setting is retained. You can also disable the keychain with the
+following command:
```
[the remark this answers is not in the document]
Comment (huyz) [reply aqua/ssh-askpass-mac/Portfile]: Fixed in https://github.com/macports/macports-ports/pull/23181/commits/0223603c328d455915c3ef99bd5db480a26e380f

**Comment by huyz** [comment]
For tracking purposes, I filed a ticket https://trac.macports.org/ticket/69581 because it appears that this can't be easily and quickly resolved.

**Comment by ryandesign** [comment]
The main.log would contain the same information with slightly different formatting.

**Comment by reneeotten** [comment]
is there any progress for this PR; if not please close until someone found the time to work on it.

**Comment by huyz** [comment]
Sure, I’ll let @ryandesign make the call since he understands better than I what the blocker is

**Comment by reneeotten** [comment]
Feel free to reopen this PR once you had the time to work on it and resolve the outstanding (CI) issues.

**Pull request closed without merging**